gem/views.py

```diff
@@ -304,3 +312,38 @@ def item_author_name(self, article_page):
 class GemAtomFeed(GemRssFeed):
     feed_type = Atom1Feed
     subtitle = GemRssFeed.description
+
+
+# https://github.com/praekelt/yal-merge/blob/develop/yal/views.py#L711-L751
+def clean_comment(self):
+    """
+    Check for email addresses and telephone numbers
+    """
+    comment = self.cleaned_data['comment']
+
+    site = Site.objects.get(is_default_site=True)
+    settings = GemSettings.for_site(site)
+
+    banned_list = [REGEX_EMAIL, REGEX_PHONE]
+
+    banned_keywords_and_patterns = \
+        settings.banned_keywords_and_patterns.split('\n') \
+        if settings.banned_keywords_and_patterns else []
+
+    banned_list += banned_keywords_and_patterns
+
+    for keyword in banned_list:
+        keyword = keyword.replace('\r', '')
+        match = re.search(keyword.lower(), comment.lower())
+        if match:
+            raise forms.ValidationError(
+                _(
+                    'This comment has been removed as it contains profanity, '
+                    'contact information or other inappropriate content. '
+                )
+            )
+
+    return comment
+
+
+CommentDetailsForm.clean_comment = clean_comment
```

Comment:
Wouldn't it be simpler to subclass the comment form? This kind of patching feels like asking for trouble.